### PR TITLE
mtu: T4572: Add DHCP-option MTU to get values from DHCP-server

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to get MTU from DHCP-server and don't touch it per
any interface change if interface `set interfaces ethernet eth1 dhcp-options mtu` is configured

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4572

## Component(s) name

mtu, dhcp
## How to test
Get DHCP MTU from DHCP-server and check that it is not changing after any other commit to default 1500:
```
set interfaces ethernet eth1 address 'dhcp'
set interfaces ethernet eth1 dhcp-options mtu
```
Check MTU:
```
vyos@r14# ip link show dev eth1
3: eth1: <BROADCAST,MULTICAST,ALLMULTI,UP,LOWER_UP> mtu 1384
```
Do any change, for example Interface Description, (expected the same MTU after commit):
```
[edit]
vyos@r14# set interfaces ethernet eth1 description VYOS
[edit]
vyos@r14# commit
[edit]
vyos@r14# ip link show dev eth1
3: eth1: <BROADCAST,MULTICAST,ALLMULTI,UP,LOWER_UP> mtu 1384
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: python/vyos/ifconfig/interface.py [hunk 1]
```diff
@@ -1535,7 +1535,7 @@ def update(self, config):
         # before mangling any IPv6 option. If MTU is less then 1280 IPv6 will be
         # automatically disabled by the kernel. Also MTU must be increased before
         # configuring any IPv6 address on the interface.
-        if 'mtu' in config:
+        if 'mtu' in config and jmespath.search('dhcp_options.mtu', config) == None:
```
Comment (c-po) [inline python/vyos/ifconfig/interface.py]: Please use `dict_search()` instead - should be faster.

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/include/interface/dhcp-options.xml.i [hunk 2]
```diff
@@ -14,6 +14,12 @@
         <help>Override system host-name sent to DHCP server</help>
       </properties>
     </leafNode>
+    <leafNode name="mtu">
+      <properties>
+        <help>Override interface MTU and get it from DHCP server</help>
```
Comment (c-po) [inline interface-definitions/include/interface/dhcp-options.xml.i]: Can we respell this: `Use MTU value from DHCP server - ignore interface setting`

**Review by c-po — CHANGES_REQUESTED** [review]

File: python/vyos/ifconfig/interface.py (hunk 1 above)
Comment (sever-sever) [reply python/vyos/ifconfig/interface.py]: > Please use `dict_search()` instead - should be faster.

Done!

File: interface-definitions/include/interface/dhcp-options.xml.i (hunk 2 above)
Comment (sever-sever) [reply interface-definitions/include/interface/dhcp-options.xml.i]: > Can we respell this: `Use MTU value from DHCP server - ignore interface setting`

Sure. Done!

**Pull request merged**